httpx/ssrf.go

```diff
@@ -64,7 +64,7 @@ func init() {
 		ssrf.WithAnyPort(),
 		ssrf.WithNetworks("tcp4", "tcp6"),
 	).Safe
-	prohibitInternalAllowIPv6 = t
+	prohibitInternalAllowIPv6 = &MeasureExternalLatencyTransport{Transport: t}
```

Comment:
Please let's not couple this functionality to the SSRF protection mechanism. We should allow users to opt into the `MeasureExternalLatencyTransport` if they wish. It's as easy as wrapping the transport at the site of `res, err := client.Do(req)`.

Reply:
This is still opt-in, because the transport is just a no-op when there is no "container" in the context. However, it does make sense to not do this per default for the httpx client.
I was at first thinking about only measuring the external latency on the "prohibitInternal*" variants of this transport, but as it is right now it does not make sense.